docs/Displaying a User Interface/Main Screen Templates/index.md

```diff
@@ -134,6 +134,294 @@ if (success) {
 ```
 !@
 
+@![iOS]
+When changing screen layouts and template data (for example, to show a weather hourly data screen vs. a daily weather screen), it is recommended to encapsulate these updates into a class or method. Doing so is a good way to keep SDL UI changes organized. A fully-formed example of this can be seen in the [example weather app](https://github.com/SmartDeviceLink-Examples/example_weather_app_ios). Below is a generic example:
+!@
+
+@![android,javaEE,javaSE]
+When changing screen layouts and template data (for example, to show a weather hourly data screen vs. a daily weather screen), it is recommended to encapsulate these updates into a class or method. Doing so is a good way to keep SDL UI changes organized. A fully-formed example of this can be seen in the [example weather app](https://github.com/SmartDeviceLink-Examples/example_weather_app_android). Below is a generic example:
+!@
+
+@![javascript]
+When changing screen layouts and template data (for example, to show a weather hourly data screen vs. a daily weather screen), it is recommended to encapsulate these updates into a class or method. Doing so is a good way to keep SDL UI changes organized. Below is a generic example:
+!@
+
+### Screen Change Protocol
+All screens will need to have access to the ```SDLScreenManager``` object and a function to display the screen. Therefore, it is recommened to create a protocol for all screens to follow. For the example below, the ```CustomSDLScreen``` protocol requires an initializer with the parameters ```SDLManager``` and a ```showScreen``` method.
```

Comment:
```suggestion
This example code creates an interface that can be implemented by various "screens" of your SDL app. This is a recommended design pattern so that you can separate your code to only involve the data models you need. This is just a simple example and your own needs may be different.

### Screen Change Example Interface
All screens will need to have access to the @![iOS]`SDLScreenManager`!@@![android, javaEE, javaSE, javascript]`ScreenManager`!@ object and a function to display the screen. Therefore, it is recommended to create a generic interface for all screens to follow. For the example below, the `CustomSDLScreen` protocol requires an initializer with the parameters `SDLManager` and a `showScreen` method.
```